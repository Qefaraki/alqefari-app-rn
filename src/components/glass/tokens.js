// Design tokens for the liquid-glass aesthetic
// Centralized so we can tune the look consistently across components

export const glassTokens = {
  radii: {
    xs: 12,
    sm: 16,
    md: 20,
    lg: 28,
    xl: 36,
  },
  blurIntensity: {
    subtle: 28,
    medium: 40,
    strong: 64,
  },
  // Base surface overlay colors for a bright UI
  surfaceOverlay: {
    // Vertical overlay from slightly brighter top to slightly dimmer bottom
    gradient: ['rgba(255,255,255,0.55)', 'rgba(255,255,255,0.28)'],
  },
  // Hairline borders that feel like condensed water on the glass
  borders: {
    inner: 'rgba(255,255,255,0.55)',
    outer: 'rgba(0,0,0,0.04)',
  },
  // Soft ambient elevation â€“ avoid harsh shadows
  elevation: {
    ios: {
      shadowColor: '#000',
      shadowOpacity: 0.10,
      shadowRadius: 14,
      shadowOffset: { width: 0, height: 8 },
    },
    android: {
      elevation: 3,
    },
  },
  highlight: {
    // Specular highlight sweeping from top-left
    sheen: ['rgba(255,255,255,0.7)', 'rgba(255,255,255,0.0)'],
  },
};

export default glassTokens;


