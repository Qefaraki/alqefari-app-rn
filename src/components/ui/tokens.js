// Design tokens for neo-native (non-glass) UI
// Neutral surfaces, soft elevation, crisp dividers, Arabic-first
// Aligned with iOS Human Interface Guidelines

const tokens = {
  colors: {
    bg: '#F2F2F7',
    surface: '#FFFFFF',
    text: '#111827',
    textMuted: '#6B7280',
    divider: 'rgba(0,0,0,0.08)',
    outline: 'rgba(0,0,0,0.12)',
    accent: '#007AFF',
    success: '#34C759',
    danger: '#FF3B30',
    najdi: {
      background: '#F9F7F3', // Al-Jass White
      container: '#D1BBA3', // Camel Hair Beige
      text: '#242121', // Sadu Night
      textMuted: '#736372',
      primary: '#A13333', // Najdi Crimson
      secondary: '#D58C4A', // Desert Ochre
      focus: '#957EB5',
    },
    diff: {
      added: '#34C759',       // iOS green for new values
      addedBg: '#34C75915',   // 8% opacity
      removed: '#FF3B30',     // iOS red for old values
      removedBg: '#FF3B3015', // 8% opacity
      modified: '#007AFF',    // iOS blue for changes
      modifiedBg: '#007AFF15', // 8% opacity
    },
  },
  // iOS-standard typography scale
  typography: {
    largeTitle: { fontSize: 34, fontWeight: '700', lineHeight: 41 },
    title1: { fontSize: 28, fontWeight: '700', lineHeight: 34 },
    title2: { fontSize: 22, fontWeight: '700', lineHeight: 28 },
    title3: { fontSize: 20, fontWeight: '600', lineHeight: 25 },
    headline: { fontSize: 17, fontWeight: '600', lineHeight: 22 },
    body: { fontSize: 17, fontWeight: '400', lineHeight: 22 },
    callout: { fontSize: 16, fontWeight: '400', lineHeight: 21 },
    subheadline: { fontSize: 15, fontWeight: '400', lineHeight: 20 },
    footnote: { fontSize: 13, fontWeight: '400', lineHeight: 18 },
    caption1: { fontSize: 12, fontWeight: '400', lineHeight: 16 },
    caption2: { fontSize: 11, fontWeight: '400', lineHeight: 13 },
  },
  radii: {
    sm: 10,
    md: 12,
    lg: 16,
    xl: 20,
    full: 9999,  // Full pill shape for chips
  },
  // iOS-standard spacing scale (8px grid)
  spacing: {
    xxs: 4,
    xs: 8,
    sm: 12,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 32,
  },
  // iOS minimum touch target
  touchTarget: {
    minimum: 44,
  },
  shadow: {
    ios: {
      shadowColor: '#000',
      shadowOpacity: 0.07,
      shadowRadius: 12,
      shadowOffset: { width: 0, height: 8 },
    },
    android: {
      elevation: 3,
    },
  },
};

export default tokens;
