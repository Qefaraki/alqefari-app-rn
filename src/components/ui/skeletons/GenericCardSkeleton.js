import React from 'react';
import { View, StyleSheet } from 'react-native';
import Shimmer from './Shimmer';
import tokens from '../tokens';

/**
 * GenericCardSkeleton - Reusable loading skeleton for InfoCard-based components
 * Used for PersonalCard, DatesCard, ProfessionalCard, ContactCard, etc.
 */
const GenericCardSkeleton = ({ rows = 3, titleWidth = 100 }) => {
  return (
    <View
      style={styles.container}
      accessible={true}
      accessibilityLabel="جاري تحميل البطاقة"
      accessibilityRole="progressbar"
    >
      {/* Title */}
      <Shimmer
        width={titleWidth}
        height={22}
        borderRadius={tokens.radii.sm}
        style={styles.title}
      />

      {/* Divider */}
      <View style={styles.divider} />

      {/* Field rows */}
      <View style={styles.body}>
        {Array.from({ length: rows }).map((_, index) => (
          <View key={`row-skeleton-${index}`} style={styles.row}>
            {/* Field label */}
            <Shimmer
              width={80 + (index * 10)}
              height={17}
              borderRadius={tokens.radii.sm}
            />
            {/* Field value (wider) */}
            <Shimmer
              width="100%"
              height={17}
              borderRadius={tokens.radii.sm}
              style={styles.value}
            />
          </View>
        ))}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: tokens.colors.najdi.background, // Al-Jass White
    borderRadius: tokens.radii.md, // 12px
    paddingHorizontal: tokens.spacing.md, // 16px
    paddingTop: tokens.spacing.md, // 16px
    paddingBottom: tokens.spacing.lg, // 20px
    marginBottom: tokens.spacing.md, // 16px

    // iOS-style elevation
    ...tokens.shadow.ios,

    // Subtle Najdi border
    borderWidth: 1,
    borderColor: tokens.colors.najdi.container + '40', // Camel Hair 40%
  },
  title: {
    marginBottom: tokens.spacing.sm, // 12px
  },
  divider: {
    height: 1,
    backgroundColor: tokens.colors.najdi.container + '20',
    marginVertical: tokens.spacing.sm, // 12px
  },
  body: {
    gap: tokens.spacing.md, // 16px between rows
  },
  row: {
    gap: tokens.spacing.xs, // 8px between label and value
  },
  value: {
    marginTop: 4,
  },
});

export default GenericCardSkeleton;
