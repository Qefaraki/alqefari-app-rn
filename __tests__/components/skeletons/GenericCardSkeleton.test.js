/**
 * GenericCardSkeleton Component Test Suite
 *
 * Tests the generic card loading skeleton used for multiple card types
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import GenericCardSkeleton from '../../../src/components/ui/skeletons/GenericCardSkeleton';
import tokens from '../../../src/components/ui/tokens';

// Mock dependencies
jest.mock('../../../src/components/ui/skeletons/Shimmer', () => 'Shimmer');

describe('GenericCardSkeleton Component', () => {
  describe('Rendering', () => {
    it('renders without crashing', () => {
      const { toJSON } = render(<GenericCardSkeleton />);
      expect(toJSON()).toBeTruthy();
    });

    it('renders with default rows (3)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      // Find all row containers
      const rows = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ gap: tokens.spacing.xs }) });

      expect(rows).toHaveLength(3);
    });

    it('renders with custom row count', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={5} />);

      const rows = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ gap: tokens.spacing.xs }) });

      expect(rows).toHaveLength(5);
    });

    it('renders with rows=1', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={1} />);

      const rows = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ gap: tokens.spacing.xs }) });

      expect(rows).toHaveLength(1);
    });

    it('renders with rows=0 (edge case)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={0} />);

      const rows = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ gap: tokens.spacing.xs }) });

      expect(rows).toHaveLength(0);
    });
  });

  describe('Card Structure', () => {
    it('renders title shimmer', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title: default width 100, height 22
      const titleShimmer = shimmers.find(s =>
        s.props.width === 100 && s.props.height === 22
      );

      expect(titleShimmer).toBeTruthy();
    });

    it('renders title with custom width', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton titleWidth={150} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      const titleShimmer = shimmers.find(s =>
        s.props.width === 150 && s.props.height === 22
      );

      expect(titleShimmer).toBeTruthy();
    });

    it('renders divider element', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const divider = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          height: 1,
          backgroundColor: tokens.colors.najdi.container + '20'
        })
      });

      expect(divider).toBeTruthy();
    });

    it('renders body container', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const body = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          gap: tokens.spacing.md
        })
      });

      expect(body).toBeTruthy();
    });
  });

  describe('Row Structure', () => {
    it('renders label shimmer for each row', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={3} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Labels: varying width (80 + index*10), height 17
      const labels = shimmers.filter(s =>
        s.props.height === 17 && typeof s.props.width === 'number'
      );

      // Should have 3 labels + 3 values
      expect(labels.length).toBeGreaterThanOrEqual(3);
    });

    it('renders value shimmer for each row', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={3} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Values: width 100%, height 17
      const values = shimmers.filter(s =>
        s.props.width === '100%' && s.props.height === 17
      );

      expect(values).toHaveLength(3);
    });

    it('varies label width progressively', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={3} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Row 0: width 80, Row 1: width 90, Row 2: width 100
      const label0 = shimmers.find(s => s.props.width === 80 && s.props.height === 17);
      const label1 = shimmers.find(s => s.props.width === 90 && s.props.height === 17);
      const label2 = shimmers.find(s => s.props.width === 100 && s.props.height === 17);

      expect(label0).toBeTruthy();
      expect(label1).toBeTruthy();
      expect(label2).toBeTruthy();
    });
  });

  describe('Design System Compliance', () => {
    it('uses Najdi background color', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          backgroundColor: tokens.colors.najdi.background
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses correct border radius (md)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          borderRadius: tokens.radii.md
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses correct padding (tokens.spacing)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          paddingHorizontal: tokens.spacing.md,
          paddingTop: tokens.spacing.md,
          paddingBottom: tokens.spacing.lg
        })
      });

      expect(container).toBeTruthy();
    });

    it('applies iOS shadow', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          shadowColor: tokens.shadow.ios.shadowColor,
          shadowOpacity: tokens.shadow.ios.shadowOpacity,
          shadowRadius: tokens.shadow.ios.shadowRadius
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses Najdi border color (40% opacity)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          borderWidth: 1,
          borderColor: tokens.colors.najdi.container + '40'
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses tokens.radii.sm for shimmers', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // All shimmers should use sm radius
      shimmers.forEach(shimmer => {
        expect(shimmer.props.borderRadius).toBe(tokens.radii.sm);
      });
    });
  });

  describe('Shimmer Count', () => {
    it('calculates correct shimmer count for 3 rows', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={3} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title(1) + (Label + Value) × 3 rows = 7
      expect(shimmers).toHaveLength(7);
    });

    it('calculates correct shimmer count for 5 rows', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={5} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title(1) + (Label + Value) × 5 rows = 11
      expect(shimmers).toHaveLength(11);
    });

    it('calculates correct shimmer count for 1 row', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={1} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title(1) + Label(1) + Value(1) = 3
      expect(shimmers).toHaveLength(3);
    });

    it('only renders title when rows=0', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={0} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Only title shimmer
      expect(shimmers).toHaveLength(1);
    });
  });

  describe('Edge Cases', () => {
    it('handles rows=0', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={0} />);
      expect(toJSON()).toBeTruthy();
    });

    it('handles negative rows (edge case)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={-1} />);
      const rows = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ gap: tokens.spacing.xs }) });

      // Negative length should create empty array
      expect(rows).toHaveLength(0);
    });

    it('handles very large row count (20 rows)', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={20} />);
      expect(toJSON()).toBeTruthy();
    });

    it('handles undefined rows (uses default 3)', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Default 3 rows: 1 title + 6 row shimmers = 7
      expect(shimmers).toHaveLength(7);
    });

    it('handles null rows', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={null} />);
      const rows = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ gap: tokens.spacing.xs }) });

      // null should be coerced to 0
      expect(rows).toHaveLength(0);
    });

    it('handles titleWidth=0', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton titleWidth={0} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      const titleShimmer = shimmers.find(s => s.props.width === 0);
      expect(titleShimmer).toBeTruthy();
    });

    it('handles very large titleWidth', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton titleWidth={500} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      const titleShimmer = shimmers.find(s => s.props.width === 500);
      expect(titleShimmer).toBeTruthy();
    });
  });

  describe('Performance', () => {
    it('renders within performance budget (< 16ms)', () => {
      const start = performance.now();
      render(<GenericCardSkeleton />);
      const end = performance.now();

      const renderTime = end - start;
      expect(renderTime).toBeLessThan(16);
    });

    it('renders 10 rows efficiently', () => {
      const start = performance.now();
      render(<GenericCardSkeleton rows={10} />);
      const end = performance.now();

      // 10 rows should still render quickly
      expect(end - start).toBeLessThan(30);
    });

    it('handles multiple instances efficiently', () => {
      const start = performance.now();

      render(
        <>
          <GenericCardSkeleton rows={3} titleWidth={80} />
          <GenericCardSkeleton rows={2} titleWidth={100} />
          <GenericCardSkeleton rows={3} titleWidth={90} />
          <GenericCardSkeleton rows={2} titleWidth={100} />
        </>
      );

      const end = performance.now();

      // 4 cards should render quickly
      expect(end - start).toBeLessThan(50);
    });
  });

  describe('Responsive Layout', () => {
    it('uses gap for row spacing', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton />);

      const body = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          gap: tokens.spacing.md
        })
      });

      expect(body).toBeTruthy();
    });

    it('uses gap between label and value', () => {
      const { UNSAFE_root } = render(<GenericCardSkeleton rows={1} />);

      const row = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          gap: tokens.spacing.xs
        })
      });

      expect(row).toBeTruthy();
    });
  });

  describe('Snapshot Tests', () => {
    it('matches snapshot with default props', () => {
      const { toJSON } = render(<GenericCardSkeleton />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('matches snapshot with 5 rows', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={5} titleWidth={120} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Use Cases', () => {
    it('works for PersonalCard (3 rows)', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={3} titleWidth={80} />);
      expect(toJSON()).toBeTruthy();
    });

    it('works for DatesCard (2 rows)', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={2} titleWidth={100} />);
      expect(toJSON()).toBeTruthy();
    });

    it('works for ProfessionalCard (3 rows)', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={3} titleWidth={90} />);
      expect(toJSON()).toBeTruthy();
    });

    it('works for ContactCard (2 rows)', () => {
      const { toJSON } = render(<GenericCardSkeleton rows={2} titleWidth={100} />);
      expect(toJSON()).toBeTruthy();
    });
  });
});
