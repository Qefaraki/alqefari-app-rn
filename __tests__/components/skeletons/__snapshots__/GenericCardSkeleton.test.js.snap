// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GenericCardSkeleton Component Snapshot Tests matches snapshot with 5 rows 1`] = `
<View
  style={
    {
      "backgroundColor": "#F9F7F3",
      "borderColor": "#D1BBA340",
      "borderRadius": 12,
      "borderWidth": 1,
      "marginBottom": 16,
      "paddingBottom": 20,
      "paddingHorizontal": 16,
      "paddingTop": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 8,
        "width": 0,
      },
      "shadowOpacity": 0.07,
      "shadowRadius": 12,
    }
  }
>
  <Shimmer
    borderRadius={10}
    height={22}
    style={
      {
        "marginBottom": 12,
      }
    }
    width={120}
  />
  <View
    style={
      {
        "backgroundColor": "#D1BBA320",
        "height": 1,
        "marginVertical": 12,
      }
    }
  />
  <View
    style={
      {
        "gap": 16,
      }
    }
  >
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={80}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={90}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={100}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={110}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={120}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
  </View>
</View>
`;

exports[`GenericCardSkeleton Component Snapshot Tests matches snapshot with default props 1`] = `
<View
  style={
    {
      "backgroundColor": "#F9F7F3",
      "borderColor": "#D1BBA340",
      "borderRadius": 12,
      "borderWidth": 1,
      "marginBottom": 16,
      "paddingBottom": 20,
      "paddingHorizontal": 16,
      "paddingTop": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 8,
        "width": 0,
      },
      "shadowOpacity": 0.07,
      "shadowRadius": 12,
    }
  }
>
  <Shimmer
    borderRadius={10}
    height={22}
    style={
      {
        "marginBottom": 12,
      }
    }
    width={100}
  />
  <View
    style={
      {
        "backgroundColor": "#D1BBA320",
        "height": 1,
        "marginVertical": 12,
      }
    }
  />
  <View
    style={
      {
        "gap": 16,
      }
    }
  >
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={80}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={90}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <Shimmer
        borderRadius={10}
        height={17}
        width={100}
      />
      <Shimmer
        borderRadius={10}
        height={17}
        style={
          {
            "marginTop": 4,
          }
        }
        width="100%"
      />
    </View>
  </View>
</View>
`;
